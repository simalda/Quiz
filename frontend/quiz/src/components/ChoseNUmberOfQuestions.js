import React from "react";

import { connect } from "react-redux";
import { onNUmberOfQuestionsSelected } from "../redux/thunkActions";
import { Link } from "react-router-dom";

class ConnectedChoseNUmberOfQuestions extends React.Component {
  handleSubmit(submitter) {
    var value = document.getElementById("numberOfQuestionsInput").value;
    submitter(value);
  }

  render() {
    return (
      <div id="container-choose">
        <div className="heade">{this.props.lang}</div>
        <div className="chooseNumberText">
          How many questions do you want in your quiz?
        </div>
        <input
          id="numberOfQuestionsInput"
          className="chooseNumberInput"
          type="text"
          name="numberOfQuestions"
        />
        <Link
          to="/quizStep"
          className="startQuiz"
          onClick={() => this.handleSubmit(this.props.chooseNumberOfQuestions)}
        >
          <div id="startButton">START THE QUIZ</div>
        </Link>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...state,
    ...ownProps,
    lang: state.languageReducer.lang
  };
};

const mapDispatchToProps = dispatch => {
  return {
    chooseNumberOfQuestions: questionNum => {
      dispatch(onNUmberOfQuestionsSelected(questionNum));
    }
  };
};

const ChoseNUmberOfQuestions = connect(
  mapStateToProps,
  mapDispatchToProps
)(ConnectedChoseNUmberOfQuestions);

export default ChoseNUmberOfQuestions;
