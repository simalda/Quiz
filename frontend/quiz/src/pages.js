const pages = {
    Login: "Login",//-1
    NumberofQuestions:"NumberofQuestions",//3
    Main: "Main",//1
    QuizStep: "QuizStep",//0
    Results:"Results",//2
    Statistics: "Statistics",//5
    AddQuestion:"AddQuestion",
    AboutUs: "AboutUs"
  };
  
  export default pages