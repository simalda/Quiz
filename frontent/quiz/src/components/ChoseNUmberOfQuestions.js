import React, { Component } from 'react';
import ReactDOM from 'react-dom';
 

class ChoseNUmberOfQuestions extends React.Component {
    constructor(props) {
        super(props)     
      }

    render(){return(
        <div >
            <div >
            How many questions do you want in your questionary?
             
            <input type="text" name="numberOfQuestions"  />  
            <button  onClick = {()=> this.props.onNUmberOfQuestionsSelected()}>Submit</button> 
            
            </div> 
       
          
   </div>
         
    );
}         }
   export default  ChoseNUmberOfQuestions