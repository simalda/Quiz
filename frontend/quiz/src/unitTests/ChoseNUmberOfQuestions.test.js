import React from "react";
import ReactDOM from "react-dom";

import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ChoseNUmberOfQuestions from "../components/ChoseNUmberOfQuestions";
import pages from "../JS/pages";

import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});
Enzyme.configure({ adapter: new Adapter() });

const state = {
  quiz: [
    {
      question: "a",
      answer: ["c", "d", "s"],
      correctAnswer: "cor"
    },
    {
      question: "ab",
      answer: ["c", "d", "s"],
      CorrectAnswer: "cor2"
    }
  ],
  highlight: "none",
  page: pages.Login,
  answerOptions: []
};

describe.skip("Choose number of questions", () => {
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(
      <Provider store={store}>
        <ChoseNUmberOfQuestions submit={mockSubmit} />
      </Provider>
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Page is correct", () => {
    expect(state.page).toBe(pages.Login);
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ChoseNUmberOfQuestions {...state} />, div);
  });

  it("should show text", () => {
    const text = wrapper.find({ id: "startButton" });
    expect(text.text()).toBe("START THE QUIZ");
  });
});
