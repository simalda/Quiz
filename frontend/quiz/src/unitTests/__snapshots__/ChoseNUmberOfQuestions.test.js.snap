// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose number of questions should match the snapshot 1`] = `ShallowWrapper {}`;
